app/sessionStore.js

```diff
@@ -447,6 +459,42 @@ module.exports.cleanSessionDataOnShutdown = () => {
   }
 }
 
+const cleanFavicons = (basePath, immutableData) => {
+  const fs = require('fs')
+  const synopsisPaths = [
+    ['ledger', 'synopsis', 'publishers'],
+    ['ledger', 'about', 'synopsis']
+  ]
+  synopsisPaths.forEach((synopsisPath) => {
+    if (immutableData.getIn(synopsisPath)) {
+      immutableData.getIn(synopsisPath).forEach((value, index) => {
+        // Fix #11582
+        if (value && value.get && isDataUrl(value.get('faviconURL', ''))) {
+          const parsed = parseFaviconDataUrl(value.get('faviconURL'))
+          if (!parsed) {
+            immutableData = immutableData.setIn(
+              synopsisPath.concat([index, 'faviconURL']), '')
+            return
+          }
+          const faviconPath = path.join(basePath,
+            typeof index === 'number'
+              ? `favicon-${index}-${Date.now()}.${parsed.ext}`
+              : `favicon-${index}.${parsed.ext}`
+          )
+          fs.writeFile(faviconPath, parsed.data, 'base64', (err) => {
+            if (err) {
+              console.error(`Error writing file: ${faviconPath} ${err}`)
```

Comment:
are we ok with potentially losing these?

Reply:
iirc the ledger tries to get the favicon whenever the user visits the page, so it would be re-fetched on the next visit

Reply:
@diracdeltas that's not the case. We only get favicon if there is not one

Reply:
i could add an img onerror in the synopsis table that sets the faviconURL to `'' / null` if the file does not exist on disk - would that cause the favicon to be re-fetched?

Reply:
yes if you set it to null it will work. More info here: https://github.com/brave/browser-laptop/blob/master/app/browser/api/ledger.js#L1146